src/main/java/com/iwaconsolti/school/miguel/service/GradeService.java

```diff
@@ -0,0 +1,143 @@
+import com.iwaconsolti.school.miguel.model.School;
+import com.iwaconsolti.school.miguel.model.Students;
+import com.iwaconsolti.school.miguel.model.dto.GradesDTO;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Slf4j
+@Service
+public class GradeService {
+
+    private final School gerardoInstitute;
+    private final School zetCollege;
+
+    @Value("${school.score.limit:100}")
+    private int limitGrade;
+
+    public GradeService(@Qualifier("GerardoInstitute") School gerardoInstitute, @Qualifier("ZetCollege") School zetCollege){
+        this.gerardoInstitute = gerardoInstitute;
+        this.zetCollege = zetCollege;
+    }
+
+    public Grade createGrade(String schoolName, GradesDTO gradeDTO) {
+        Grade grade = new Grade();
+        grade.setId(gradeDTO.getId());
+        grade.setScore(gradeDTO.getScore());
+
+        Students student = findStudentById(schoolName, gradeDTO.getStudentId());
+        Courses course = findCourseById(schoolName,gradeDTO.getCourseId());
+
+        if("GerardoInstitute".equalsIgnoreCase(schoolName) && grade.getScore() <= limitGrade){
+            grade.setStudent(student);
+            grade.setCourse(course);
+
+            gerardoInstitute.getGrades().put(grade.getId(), grade);
+            log.info("The grade was successfully registered in Gerardo Institute {}",grade);
+        }else if("ZetCollege".equalsIgnoreCase(schoolName) && grade.getScore() <= limitGrade) {
+            zetCollege.getGrades().put(grade.getId(), grade);
+            log.info("The grade was successfully registered in Zet College {}",grade);
+        }
+        return grade;
+    }
+
+    public List<String> getGradesByStudentId(String schoolName, Integer id){
+        List<String> studentGrades = new ArrayList<>();
+        if("GerardoInstitute".equalsIgnoreCase(schoolName)){
+            for(Grade grade : gerardoInstitute.getGrades().values()) {
+                if (grade.getStudent().getId() == id) {
+                    studentGrades.add("Name: " + grade.getStudent().getFirstName() + " " + grade.getStudent().getLastName());
+                    studentGrades.add("Age: " + grade.getStudent().getAge());
+                    studentGrades.add("Professor: " + grade.getCourse().getProfessorName());
+                    studentGrades.add("Curse: " + grade.getCourse().getNameCourse());
+                    studentGrades.add("Score: " + grade.getScore());
+                }
+            }
+        }else if("ZetCollege".equalsIgnoreCase(schoolName)) {
+            for(Grade grade: zetCollege.getGrades().values()) {
+                if (grade.getStudent().getId() == id) {
+                    studentGrades.add("Name: " + grade.getStudent().getFirstName() + " " + grade.getStudent().getLastName());
```

Comment:
This and the code above is duplicated, it could have being extracted to a private method